core/src/test/java/org/stellar/anchor/Sep6ServiceTest.kt

```diff
@@ -122,4 +135,242 @@ class Sep6ServiceTest {
+    }
+
+    verify { txnStore.findByTransactionId(any()) }
+  }
+
+  @Test
+  fun `test find transaction belonging to different account`() {
+    val depositTxn = createDepositTxn("other-account")
+    val request = GetTransactionRequest.builder().id(depositTxn.id).lang("en-US").build()
+    every { txnStore.findByTransactionId(depositTxn.id) } returns depositTxn
+
+    assertThrows<NotFoundException> {
+      sep6Service.findTransaction(TestHelper.createSep10Jwt(TEST_ACCOUNT), request)
+    }
+
+    verify { txnStore.findByTransactionId(depositTxn.id) }
+  }
+
+  @Test
+  fun `test find transaction with different account memo`() {
+    val depositTxn = createDepositTxn(TEST_ACCOUNT, "other-memo")
+    val request = GetTransactionRequest.builder().id(depositTxn.id).lang("en-US").build()
+    every { txnStore.findByTransactionId(depositTxn.id) } returns depositTxn
+
+    assertThrows<NotFoundException> {
+      sep6Service.findTransaction(TestHelper.createSep10Jwt(TEST_ACCOUNT), request)
+    }
+
+    verify { txnStore.findByTransactionId(depositTxn.id) }
+  }
+
+  @Test
+  fun `test find transactions with account mismatch`() {
+    val request =
+      GetTransactionsRequest.builder()
+        .assetCode(TEST_ASSET)
+        .account(TEST_ACCOUNT)
+        .noOlderThan(Instant.now().toString())
+        .limit(10)
+        .kind("deposit")
+        .pagingId("1")
+        .lang("en-US")
+        .build()
+
+    assertThrows<SepNotAuthorizedException> {
+      sep6Service.findTransactions(TestHelper.createSep10Jwt("other-account"), request)
+    }
+    verify { txnStore wasNot Called }
+  }
+
+  @Test
+  fun `test find transactions with unsupported asset`() {
+    val request =
+      GetTransactionsRequest.builder()
+        .assetCode("???")
+        .account(TEST_ACCOUNT)
+        .noOlderThan(Instant.now().toString())
+        .limit(10)
+        .kind("deposit")
+        .pagingId("1")
+        .lang("en-US")
+        .build()
+
+    assertThrows<SepValidationException> {
+      sep6Service.findTransactions(TestHelper.createSep10Jwt(TEST_ACCOUNT), request)
+    }
+    verify { txnStore wasNot Called }
+  }
+
+  @Test
+  fun `test find transactions`() {
+    val depositTxn = createDepositTxn(TEST_ACCOUNT)
+    every { txnStore.findTransactions(TEST_ACCOUNT, any(), any()) } returns listOf(depositTxn)
+    val request =
+      GetTransactionsRequest.builder()
+        .assetCode(TEST_ASSET)
+        .account(TEST_ACCOUNT)
+        .noOlderThan(Instant.now().toString())
+        .limit(10)
+        .kind("deposit")
+        .pagingId("1")
+        .lang("en-US")
+        .build()
+    val response = sep6Service.findTransactions(TestHelper.createSep10Jwt(TEST_ACCOUNT), request)
+
+    verify(exactly = 1) { txnStore.findTransactions(TEST_ACCOUNT, null, request) }
+
+    assert(response.transactions.size == 1)
+    assertEquals(depositTxn.id, response.transactions[0].id)
```

Comment:
Can we use JSONAssert?

Reply:
I think this test is more robust as we are asserting the values in the transaction response came from the deposit transaction. With JSONAssert, we only assert that the transaction response fields have certain values.